### ISSUE TITLE
virtual layer left join incomplete results

### ISSUE THREAD

**Issue by CommunyAS** [qgis/QGIS#54447]
### What is the bug or the crash?

I get incomplete results when using `left join` to create a virtual layers from two separate layers A and B.

**A** contains basic information with field `ID`defined as INTEGER PRIMARY KEY.
**B** contains contains additional information, with multiple entries for every entry in A, a field `ID`to relate to **A** exists - i.e. both stand in a 1:n relation.

When relating both **A** and **B** like
``` SQL
select *
from A
left join B
    on A.ID = B.ID
```

The number of entries in the virtual layer is equal to the number of entries in **A**, but of cause should be much higher (in my case, since the is not entry in **B** not relating to an entry in **A**, it should be the number of entries in **B**).

It looks like that the `left join' clause on a 1:n relation reacts like or emulates a 1:1, probably because i.e. only the first matching entry in **B** is selected.

A check with DBeaver script using the same code produced the correct results (i.e. in my case return the entries from **A** and all related entries from **B**).

Perhaps there is a relation to another bug report [virtual layer join wrong results #54177](https://github.com/qgis/QGIS/issues/54177).

### Steps to reproduce the issue

1. create 2 layers as described above with a field `ID` to relate them.
2. create a virtual layer with SQL code described above.
3. see incomplete results

### Versions

QGIS version | 3.28.10-Firenze | QGIS code revision | 09a302a6ca5
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.5
GDAL/OGR version | 3.3.2
PROJ version | 8.1.1
EPSG Registry database version | v10.028 (2021-07-07)
GEOS version | 3.9.1-CAPI-1.14.2
SQLite version | 3.35.2
PDAL version | 2.3.0
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.11.5
OS version | macOS 13.4
  |   |   |  
Active Python plugins
processing | 2.12.99
sagaprovider | 2.12.99
grassprovider | 2.12.99
db_manager | 0.1.20
MetaSearch | 0.3.6

QGIS version
3.28.10-Firenze
QGIS code revision
[09a302a6ca5](https://github.com/qgis/QGIS/commit/09a302a6ca5)
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.3.2
PROJ version
8.1.1
EPSG Registry database version
v10.028 (2021-07-07)
GEOS version
3.9.1-CAPI-1.14.2
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.11.5
OS version
macOS 13.4

Active Python plugins
processing
2.12.99
sagaprovider
2.12.99
grassprovider
2.12.99
db_manager
0.1.20
MetaSearch
0.3.6

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by alexbruy** [comment]
Duplicate of #54177?

**Comment by CommunyAS** [comment]
No, not a duplicate to #54177 - what is described there differs from what I describe in my initial post. But as mentioned above: **Perhaps there is a relation to another bug report [virtual layer join wrong results #54177](https://github.com/qgis/QGIS/issues/54177).**
